server/routes/teacherRouter.js

```diff
@@ -18,6 +19,24 @@ teacherRouter.get("/login", async (req, res) => {
   }
 })
 
+teacherRouter.get("/courses", async (req, res) => {
+  try {
+    const profId = req.query.prof_id; // getting the  prof id from the request query
+    if (!profId) {
+      return res.status(400).json({ message: "Prof id required" });
+    }
+
+    // Find courses by prof id
+    const courses = await Course.find({ prof_id: profId });
+    res.json(courses);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
+  }
+});
```

Comment:
A lot of validation is missing please referrer to the `/register` route

Reply:
Thank you for the feedback! I have committed the changes